File: LeetCode/ArrayStrings/python/romanToInt.py
class Solution:
    def romanToInt(self, s: str) -> int:
        romanToIntMap = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}

        res = 0

        for i in range(0, len(s)):
            
            s2 = s[i+1] if i < len(s) - 1 else ""

            if not s2 == "" and romanToIntMap[s[i]] < romanToIntMap[s[i+1]]:
                res -= romanToIntMap[s[i]]
            else:
                res += romanToIntMap[s[i]]

        return res

    @staticmethod
    def testSolution(input) -> None:
        print("Input: s: {}".format(input["s"]))
        print("Expected: {}".format(input["expected"]))
        print("Result: {}".format(Solution().romanToInt(input["s"])))
        print("-" * 50)

if __name__ == "__main__":
    input = {"s": "III", "expected": 3}
    Solution.testSolution(input)
    
    input = {"s": "LVIII", "expected": 58}
    Solution.testSolution(input)

    input = {"s": "MCMXCIV", "expected": 1994}
    Solution.testSolution(input)

File: LeetCode/ArrayStrings/java/romanToInt.java
package LeetCode.ArrayStrings.java;

enum romanToIntEnum {

    I(1),
    V(5),
    X(10),
    L(50),
    C(100),
    D(500),
    M(1000);

    private final int value;

    romanToIntEnum(int value) {
        this.value = value;
    }

    public int getValue() {
        return this.value;
    }
}

record romanToIntRecord(String s, int expected) {
}

public class romanToInt {
    public int romanToIntSoln(String s) {
        int res = 0;

        for (int i = 0; i < s.length(); i += 1) {
            // i is less than s.length - 1, perform res -=
            // else perform res +=
            String s1 = String.valueOf(s.charAt(i));
            String s2 = i < s.length() - 1 ? String.valueOf(s.charAt(i + 1)) : "";

            if (!s2.equals("") &&
                    romanToIntEnum.valueOf(s1).getValue() < romanToIntEnum.valueOf(s2).getValue()) {

                res -= romanToIntEnum.valueOf(s1).getValue();
            } else {
                res += romanToIntEnum.valueOf(s1).getValue();
            }
        }

        return res;
    }

    public static void main(String[] args) {
        romanToIntRecord input = new romanToIntRecord("III", 3);
        romanToInt.testSolution(input);

        input = new romanToIntRecord("LVIII", 58);
        romanToInt.testSolution(input);

        input = new romanToIntRecord("MCMXCIV", 1994);
        romanToInt.testSolution(input);
    }

    private static void testSolution(romanToIntRecord input) {
        System.out.println("Input: s: " + input.s());
        System.out.println("Expected " + input.expected());
        System.out.println("Result: " + new romanToInt().romanToIntSoln(input.s()));
        System.out.println("-".repeat(50));
    }
}
